cpp/oneapi/dal/io/detail/load_graph.hpp

```diff
@@ -53,126 +54,127 @@ edge_list<std::int32_t> load_edge_list(const std::string &name) {
-    using allocator_atomic_t =
-        typename std::allocator_traits<allocator_t>::template rebind_alloc<atomic_t>;
+    auto layout = oneapi::dal::preview::detail::get_impl(g);
+    auto &allocator = layout->_allocator;
+    layout->_vertex_count = n_vertex;
+
+    void *degrees_vec_void =
+        (void *)allocator.allocate(n_vertex * (sizeof(atomic_t) / sizeof(char)));
+    atomic_t *degrees_vec = new (degrees_vec_void) atomic_t[n_vertex];
 
-    auto *degrees_vec = new (std::nothrow)
-        oneapi::dal::preview::detail::graph_container<atomic_t, allocator_atomic_t>(
-            _unf_vertex_count);
     if (degrees_vec == nullptr) {
         throw bad_alloc();
     }
-    atomic_t *degrees_cv = degrees_vec->data();
+    atomic_t *degrees_cv = degrees_vec;
     if (degrees_cv == nullptr) {
         throw bad_alloc();
     }
 
-    threader_for(_unf_vertex_count, _unf_vertex_count, [&](vertex_t u) {
-        degrees_cv[u].set(0);
-    });
-
     threader_for(edges.size(), edges.size(), [&](vertex_t u) {
         degrees_cv[edges[u].first].inc();
         degrees_cv[edges[u].second].inc();
     });
 
-    auto *rows_vec = new (std::nothrow)
-        oneapi::dal::preview::detail::graph_container<atomic_t, allocator_atomic_t>(
-            _unf_vertex_count + 1);
+    void *rows_vec_void =
+        (void *)allocator.allocate((n_vertex + 1) * (sizeof(atomic_t) / sizeof(char)));
+    atomic_t *rows_vec = new (rows_vec_void) atomic_t[n_vertex + 1];
+
     if (rows_vec == nullptr) {
         throw bad_alloc();
     }
-    atomic_t *rows_cv = rows_vec->data();
+    atomic_t *rows_cv = rows_vec;
     if (rows_cv == nullptr) {
         throw bad_alloc();
     }
 
     vertex_t total_sum_degrees = 0;
     rows_cv[0].set(total_sum_degrees);
-
-    for (vertex_t i = 0; i < _unf_vertex_count; ++i) {
+    for (vertex_t i = 0; i < n_vertex; ++i) {
         total_sum_degrees += degrees_cv[i].get();
         rows_cv[i + 1].set(total_sum_degrees);
     }
-    delete degrees_vec;
+    allocator.deallocate((char *)degrees_vec_void, n_vertex * (sizeof(atomic_t) / sizeof(char)));
+
+    void *_unf_vert_neighs_vec_void =
+        (void *)allocator.allocate((rows_cv[n_vertex].get()) * (sizeof(vertex_t) / sizeof(char)));
+    vertex_t *_unf_vert_neighs_arr =
+        new (_unf_vert_neighs_vec_void) vertex_t[rows_cv[n_vertex].get()];
 
-    vector_vertex_t _unf_vert_neighs_vec(rows_cv[_unf_vertex_count].get());
-    vector_edge_t _unf_edge_offset_vec(_unf_vertex_count + 1);
-    auto _unf_edge_offset_arr = _unf_edge_offset_vec.data();
-    auto _unf_vert_neighs_arr = _unf_vert_neighs_vec.data();
+    void *_unf_edge_offset_arr_void =
+        (void *)allocator.allocate((n_vertex + 1) * (sizeof(edge_t) / sizeof(char)));
+    edge_t *_unf_edge_offset_arr = new (_unf_edge_offset_arr_void) edge_t[n_vertex + 1];
 
-    threader_for(_unf_vertex_count + 1, _unf_vertex_count + 1, [&](vertex_t n) {
+    threader_for(n_vertex + 1, n_vertex + 1, [&](vertex_t n) {
```

Comment:
If you use old `threaded_for`, I suggest to move this implementation to backend cpp file not to expose DAAL headers from oneDAL. Moreover, it is better to create wrapper over `threaded_for` into interop or implement it by scratch in new interfaces

Reply:
ok